lib/pbio/drv/core.c

```diff
@@ -92,8 +92,10 @@ void pbdrv_init(void) {
     pbdrv_usb_init();
     pbdrv_watchdog_init();
 
-    // Some hubs have a bootloader that disables interrupts. Has to be done
-    // here otherwise Essential hub can hang on boot if it is earlier.
+    // Some hubs have a bootloader that disables interrupts, and on other hubs
+    // we disable them early in the boot process to avoid corrupting statically
```

Comment:
Could be simpler to just say that "interrupts are disabled when transitioning from the bootlaoder to our firmware" and leave it at that. (The bit about corrupting memory is just one of several possible side effects but isn't really relevant here. The really important information here is the last sentence explaining why we can't enable interrupts any sooner than this.)

Reply:
Thanks, will update. But to nitpick further I couldn't quite follow the existing comment on why it was there and not before, or perhaps after setting up pbsys, etc.

:smile:

Reply:
Fair point. :smile: 

I never tracked it down to the exact problem so it is a bit vague. But basically, if an interrupt happens before the first init of all of the drivers is called, it can cause problems. But they need to be enabled for drivers with init_up/down waiting to be able to finish initializing.